fuel-vm/src/predicate.rs

```diff
@@ -123,3 +128,30 @@ fn from_tx_works() {
         assert_eq!(program, &padded_predicate);
     }
 }
+
+/// Extension trait that allows checking predicates of [`Checked<Tx>`]
+pub trait CheckedTxExt
+where
+    Self: Sized,
+{
+    /// Check predicates, if not yet done.
+    fn check_predicates(self, params: ConsensusParameters, gas_costs: GasCosts) -> Result<Self, CheckError>;
+}
+
+impl<Tx: IntoChecked + ExecutableTransaction + CheckedMetadata> CheckedTxExt for Checked<Tx>
+where
+    Self: Clone,
+    <Tx as IntoChecked>::Metadata: CheckedMetadata,
+{
+    /// Check predicates, if not yet done.
+    fn check_predicates(mut self, params: ConsensusParameters, gas_costs: GasCosts) -> Result<Self, CheckError> {
+        if !self.checks().contains(Checks::Predicates) {
+            if Interpreter::<PredicateStorage>::check_predicates(self.clone(), params, gas_costs) {
```

Comment:
It is kind of confusing that the boolean returned by `check_predicates` indicates a failure if `true`? Or am I misreading this check?

Reply:
Changed the return type to a `Result`, so that it's more readable